### PR TITLE
Only build netfx package on allconfigurations

### PR DESCRIPTION
**Pull request by joperezr**
cc: @weshaggard @ericstj @MattGal 

Fixing netfx official build by:
 - Only producing the new netfx package on the AllConfigurations leg
 - Skipping the publish step if there are no packages to publish

### CODE REVIEW

File: src/publish.proj [hunk 1]
```diff
@@ -8,6 +8,15 @@
...
   </PropertyGroup>
 
+  <ItemGroup>
+    <_PackagesToPublish Include="$(PublishPattern)" />
+  </ItemGroup>
+
+  <PropertyGroup>
+    <!-- only try to publish if there are .nupkgs produced -->
+    <PublishBuildDependsOn Condition="'@(_PackagesToPublish)' != ''">CreateContainerName;UploadToAzure</PublishBuildDependsOn>
+  </PropertyGroup>
```
Comment (weshaggard) [inline src/publish.proj]: Can you please file a tracking issue for this?

File: src/publish.proj [hunk 2]
```diff
@@ -8,6 +8,15 @@
     <PublishPattern Condition="'$(PublishPattern)' == ''">$(PackageOutputRoot)**\*.nupkg</PublishPattern>
   </PropertyGroup>
 
+  <ItemGroup>
+    <_PackagesToPublish Include="$(PublishPattern)" />
+  </ItemGroup>
+
+  <PropertyGroup>
+    <!-- only try to publish if there are .nupkgs produced -->
+    <PublishBuildDependsOn Condition="'@(_PackagesToPublish)' != ''">CreateContainerName;UploadToAzure</PublishBuildDependsOn>
```
Comment (ericstj) [inline src/publish.proj]: I don't think this works.  Properties outside a target are evaluated before items.  https://msdn.microsoft.com/en-us/library/dd997067.aspx#Anchor_2

I would have expected this to error with
```
error MSB4099: A reference to an item list at position n is not allowed in this condition
```

Comment (weshaggard) [inline src/publish.proj]: If that is the case please just move it into a target.

Comment (joperezr) [inline src/publish.proj]: will do

**Pull request merged**